sdk/core/core-http/lib/policies/keepAlivePolicy.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { BaseRequestPolicy, RequestPolicy, RequestPolicyOptions } from './requestPolicy';
+import { WebResource } from '../webResource';
+import { HttpOperationResponse } from '../httpOperationResponse';
+
+/**
+ * Options for how HTTP connections should be maintained for future
+ * requests.
+ */
+export interface KeepAliveOptions {
+  /*
+   * When true, connections will be kept alive for multiple requests.
+   * Defaults to true.
+   */
+  enable: boolean;
+
+  /*
+   * The maximum number of sockets that will be opened per host.
+   * Defaults to 50.
+   */
+  maxSockets?: number;
+}
+
+export const DefaultKeepAliveOptions: KeepAliveOptions = {
+  enable: true,
+  maxSockets: 50
+}
+
+export function keepAlivePolicy(keepAliveOptions?: KeepAliveOptions) {
+  return {
+    create: (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
+      return new KeepAlivePolicy(nextPolicy, options, keepAliveOptions || DefaultKeepAliveOptions);
+    }
+  };
+}
+
+/**
+ * KeepAlivePolicy is a policy used to control keep alive settings for every request.
+ */
+export class KeepAlivePolicy extends BaseRequestPolicy {
+  /**
+   * Creates an instance of KeepAlivePolicy.
+   *
+   * @param {RequestPolicy} nextPolicy
+   * @param {RequestPolicyOptions} options
+   * @param {KeepAliveOptions} [keepAliveOptions]
+   */
+  constructor(
+    nextPolicy: RequestPolicy,
+    options: RequestPolicyOptions,
+    private readonly keepAliveOptions: KeepAliveOptions
+  ) {
+    super(nextPolicy, options);
+  }
+
+  /**
+   * Sends out request.
+   *
+   * @param {WebResource} request
+   * @returns {Promise<HttpOperationResponse>}
+   * @memberof KeepAlivePolicy
+   */
+  public async sendRequest(request: WebResource): Promise<HttpOperationResponse> {
+    request.keepAlive = this.keepAliveOptions.enable;
```

Comment:
Don't we also need to set `maxSockets` on the request from the `this.keepAliveOptions`?

Reply:
Yep, thanks for catching that!

Reply:
Removed this setting for now because it'll take more work to route it to the right place.